components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -308,6 +365,92 @@ static esp_err_t set_websocket_transport_optional_settings(esp_websocket_client_
     return ESP_ERR_INVALID_ARG;
 }
 
+static esp_err_t esp_websocket_client_create_transport(esp_websocket_client_handle_t client)
+{
+    if (client->transport_list) {
+        esp_transport_list_destroy(client->transport_list);
+        client->transport_list = NULL;
+    }
+
+    if (client->config->scheme) {
+        client->transport_list = esp_transport_list_init();
+        ESP_WS_CLIENT_MEM_CHECK(TAG, client->transport_list, return ESP_ERR_NO_MEM);
+        if (strcasecmp(client->config->scheme, WS_OVER_TCP_SCHEME) == 0) {
+            esp_transport_handle_t tcp = esp_transport_tcp_init();
+            ESP_WS_CLIENT_MEM_CHECK(TAG, tcp, return ESP_ERR_NO_MEM);
+
+            esp_transport_set_default_port(tcp, WEBSOCKET_TCP_DEFAULT_PORT);
+            esp_transport_list_add(client->transport_list, tcp, "_tcp"); // need to save to transport list, for cleanup
+            if (client->keep_alive_cfg.keep_alive_enable) {
+                esp_transport_tcp_set_keep_alive(tcp, &client->keep_alive_cfg);
+            }
+            if (client->if_name) {
+                esp_transport_tcp_set_interface_name(tcp, client->if_name);
+            }
+
+            esp_transport_handle_t ws = esp_transport_ws_init(tcp);
+            ESP_WS_CLIENT_MEM_CHECK(TAG, ws, {
+                esp_transport_destroy(tcp);
```

Comment:
Must call `esp_transport_list_destroy` to destroy both `tcp` and the list.

Reply:
I think we can return `ESP_ERR_NO_MEM` directly and `esp_websocket_client_destory` will call `esp_transport_list_destroy` to free all the transport that added to list.